syntax = "proto3";

import "common.proto";
package autodrive.workflow.proto;

enum ProcessorStatus {
  kProcessorStatusCreated = 0;
  kProcessorStatusNormal = 1;
  kProcessorStatusException = 2;
  kProcessorStatusError = 3;
  kProcessorStatusDestroying = 4;
  kProcessorStatusDestroyed = 5;
}

enum ProcessorErrorCode {
  kProcessorErrorCodeNone = 0;
  kProcessorErrorCodeImageSyncFailed = 1;
  kProcessorErrorCodeOdoCheckFailed = 2;
  kProcessorErrorCodeCamFilesReadFailed = 3;
  kProcessorErrorCodeOdoTimestampFailed = 4;
  kStaticObsCheckFailed = 5; 
  kDynamicObsCheckFailed = 6;
  kUssObsCheckFailed = 7;
  kFrontObsCheckFailed = 8;
  kParkingSlotsCheckFailed = 9;
  kCylinderParsingCheckFailed = 10;
  kIpmParsingCheckFailed = 11;
  kUssParkingSlotsCheckFailed = 12;
  kCameraOnlineCheckFaliled = 13;
}

message ProcessorInfoMsg {
  // Header
  MsgHeader header = 1;
  // Error
  Error error = 2;
  ProcessorStatus status = 3;
  ProcessorErrorCode error_code = 4;
}